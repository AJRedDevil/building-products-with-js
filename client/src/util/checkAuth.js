const key = 'user.token';

export const isAuthenticated = () => localStorage.getItem(key);
