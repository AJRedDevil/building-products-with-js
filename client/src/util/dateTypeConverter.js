import moment from 'moment';

export const convertStringToDate = strDate => moment(strDate).toDate();
export const convertDateToISOString = dateType => moment(dateType).toISOString();
