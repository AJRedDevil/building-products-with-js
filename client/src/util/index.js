export {PrivateRoute} from './requireAuth';
export {isAuthenticated} from './checkAuth';
export {
  loginErrorToMessage, registerErrorToMessage, ajaxErrorToMessage,
} from './errorToMessage';
export {signRequest} from './signRequest';
export {MyPropType} from './ourPropTypes';
