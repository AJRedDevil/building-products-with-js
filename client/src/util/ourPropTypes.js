import PropTypes from 'prop-types';

const UserPropType = PropTypes.shape({
  id: PropTypes.String,
  login: PropTypes.String,
  registrationDate: PropTypes.string,
});

const UserDefaultProp = {
  id: '',
  login: '',
  registrationDate: '',
};

const QuestionPropType = PropTypes.shape({
  answers: PropTypes.array,
  creationDate: PropTypes.string,
  expirationDate: PropTypes.string,
  id: PropTypes.string,
  owner: UserPropType,
  text: PropTypes.string,
});

const QuestionDefaultProp = {
  answers: [],
  creationDate: '',
  expirationDate: '',
  id: '',
  owner: UserDefaultProp,
  text: '',
};

const NotificationPropType = PropTypes.shape({
  id: PropTypes.number,
  alertType: PropTypes.string,
  text: PropTypes.string,
});

const NotificationDefaultProp = {
  id: 0,
  alertType: '',
  text: '',
};

const EmptyFuncDefaultProp = () => {}

export const MyPropType = {
  EmptyFuncDefaultProp,
  QuestionPropType,
  QuestionDefaultProp,
  NotificationPropType,
  NotificationDefaultProp,
  UserPropType,
  UserDefaultProp,
};
