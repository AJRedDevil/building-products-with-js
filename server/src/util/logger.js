import winston from 'winston';

let level;
switch (process.env.NODE_ENV) {
  case 'testing':
    level = 'error';
    break;
  case 'production':
    level = 'info';
    break;
  default:
    level = 'debug';
    break;
}

export const logger = new winston.Logger({
  transports: [
    new winston.transports.Console({
      level,
      colorize: true,
      timestamp: true,
      prettyPrint: true,
      label: 'experts-server',
    }),
  ],
});

// create stream for morgan
logger.stream = {
  write: message => logger.info(message),
};

export {logger as default};
