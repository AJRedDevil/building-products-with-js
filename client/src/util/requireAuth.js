// npm packages
import React from 'react';
import {Redirect, Route} from 'react-router-dom';
import PropTypes from 'prop-types';

// our packages
import {isAuthenticated} from './checkAuth';

export const PrivateRoute = ({component: Component, ...rest}) => (
  <Route
    {...rest}
    render={props => (
      isAuthenticated() ? (
        <Component {...props} />
      ) : (
        <Redirect to={{
          pathname: '/login',
          state: {from: props.location.pathname},
          }}
        />
      )
    )
    }
  />
);
PrivateRoute.propTypes = {
  component: PropTypes.func.isRequired,
  location: PropTypes.shape({pathname: PropTypes.string}),
};
PrivateRoute.defaultProps = {
  location: {pathname: ''},
};
