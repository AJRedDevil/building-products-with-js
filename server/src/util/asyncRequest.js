import {logger} from './logger';

const asyncRequest = handler =>
  (req, res) =>
    handler(req, res).catch(e => logger.error('error diring request', e));

export default asyncRequest;
