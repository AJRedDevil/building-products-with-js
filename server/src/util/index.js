import hash from './hash';
import asyncRequest from './asyncRequest';

export {logger} from './logger';
export {asyncRequest, hash};
