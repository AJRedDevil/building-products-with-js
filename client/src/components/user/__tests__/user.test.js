// npm packages
import React from 'react';
import configureMockStore from 'redux-mock-store';

// our packages
import UserWrapper, {User} from '../index';
import {convertDateToISOString} from '../../../util';

// create mockstore
const mockStore = configureMockStore();

const user = {
  id: 0,
  login: 'test',
  registrationDate: convertDateToISOString(new Date(2017, 1, 1, 1, 1, 1)),
};

test('# UserWrapper', () => {
  const store = mockStore();
  const wrapper = shallow(<UserWrapper user={user} store={store} />);
  expect(wrapper).toMatchSnapshot();
});

test('# User', () => {
  const newLogin = 'newLogin';
  const updateUser = ({login}) => expect(login).toBe(newLogin);

  const component = (
    <User
      user={user}
      updateUser={updateUser}
      edit
    />
  );
  // test rendering
  const wrapper = shallow(component);
  expect(wrapper).toMatchSnapshot();

  // test user update
  const app = mount(component);
  // set new usernane
  app.find('#userInput').getDOMNode().value = newLogin;
  // click answer button
  app.find('button').simulate('click');
});
