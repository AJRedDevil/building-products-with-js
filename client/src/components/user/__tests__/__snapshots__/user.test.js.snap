// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`# User 1`] = `
<div
  className="panel panel-default"
  key="0"
>
  <div
    className="panel-heading"
  >
    User: 
    <input
      defaultValue="test"
      id="userInput"
      type="text"
    />
    <div
      className="pull-right"
    >
      <button
        className="btn btn-default"
        onClick={[Function]}
      >
        Save
      </button>
    </div>
  </div>
  <div
    className="panel-body"
  >
    Registration date: 
    Wed Feb 01 2017 01:01:01 GMT+0545
  </div>
</div>
`;

exports[`# UserWrapper 1`] = `
<User
  edit={false}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  updateUser={[Function]}
  user={
    Object {
      "id": 0,
      "login": "test",
      "registrationDate": "2017-01-31T19:16:01.000Z",
    }
  }
/>
`;
